test/reply-error.test.js

```diff
@@ -373,7 +373,7 @@ test('should set the status code and the headers from the error object (from cus
 
   fastify.setErrorHandler((err, request, reply) => {
     t.is(err.message, 'ouch')
-    t.is(reply.raw.statusCode, 401)
+    t.is(reply.raw.statusCode, 200)
```

Comment:
This change is expected? Seems is expected 401 here since the `Error.statusCode = 401`

Reply:
The new intended behavior is that the statusCode shouldn't be set for a custom error handler. It would only be set if the custom error handler returns a new error or if there is no custom error handler.

Reply:
So, the test title needs to change, right?

Reply:
No I don't think so. The error object sent from the custom error handler should still set the status code and header for the response. The new change is that the reply object shouldn't already be touched before going to the custom error handler. I have a new test that checks for that.